src/drivers/mct/main/component_mod.F90

```diff
@@ -892,4 +898,41 @@ subroutine component_diag(infodata, comp, flow, comment, info_debug, timer_diag
 
   end subroutine component_diag
 
+  subroutine check_fields(comp, comp_index)
+    use shr_infnan_mod, only: shr_infnan_isnan
+    use mct_mod, only: mct_avect_getrlist2c, mct_gsMap_orderedPoints
+    type(component_type), intent(in) :: comp
+    integer(in), intent(in) :: comp_index
+
+    integer(IN)   :: lsize             ! size of attr vect
+    integer(IN)   :: nflds             ! number of attr vects
+    integer(IN)   :: nancnt
+    integer(in)   :: fld, n            ! iterators
+    integer(IN)  :: rank
+    integer(IN) :: ierr
+    integer(IN), pointer :: gpts(:)
+    character(len=CL) :: msg
+    real(r8), pointer :: rattr(:,:)
+
+    lsize = mct_avect_lsize(comp%c2x_cc)
+    nflds = size(comp%c2x_cc%rattr,1)
+    if(lsize <= 1 .and. nflds <= 1) return
```

Comment:
The c2x_cc structure is allocated even if it is not used (stub components) in this case size is  1,1 and the field should not be tested.